Agent diagnostics (#5254)
elastic/docs-content: troubleshoot/ingest/fleet/diagnostic.md

1. Enable [`debug` logging](/reference/fleet/monitor-elastic-agent.md#change-logging-level).
2. [Restart {{agent}}](/reference/fleet/agent-command-reference.md#elastic-agent-restart-command).
3. Wait 10 minutes for changes to sync from {{fleet}} server to {{agent}} and for it to restart.
4. Pull the {{agent}} diagnostic using prefered method.
5. Disable `debug` logging.
:::

## Collect {{agent}} diagnostics [agent-diagnostics-collect] 

{{agent}} comes bundled with a [`diagnostics` command](/reference/fleet/agent-command-reference.md#elastic-agent-diagnostics-command) which generates a zip archive containing troubleshooting diagnostic information. This export is intended only for debugging purposes and its structure can change between releases.

The {{fleet}} UI provides the ability to remotely generate and gather an {{agent}}'s diagnostics bundle if it is online in a [`Healthy` or `Unhealthy` status](/reference/fleet/monitor-elastic-agent.md#view-agent-status). For {{agent}}s in other statuses, you must use the CLI to grab their diagnostic.

:::{warning}

> Diagnostics and logs mainly emit product metadata and settings, but they may expose sensitive data which needs to be redacted before being shared outside of your organization. See [Sanitize](#agent-diagnostics-sanitize)

Author: ```suggestion
Diagnostics and logs mainly emit product metadata and settings, but they might expose sensitive data which needs to be redacted before being shared outside of your organization. Refer to [Sanitize](#agent-diagnostics-sanitize)
```
